File: src/main/java/com/heybys/oddments/fooddelivery/domain/shop/OptionGroupId.java
package com.heybys.oddments.fooddelivery.domain.shop;

import com.heybys.oddments.base.domain.LongTypeIdentifier;
import com.heybys.oddments.base.jpa.hibernate.LongTypeIdentifierJavaType;

public class OptionGroupId extends LongTypeIdentifier {

    protected OptionGroupId(Long id) {
        super(id);
    }

    public static class OptionGroupIdJavaType extends LongTypeIdentifierJavaType<OptionGroupId> {
        public OptionGroupIdJavaType() {
            super(OptionGroupId.class);
        }
    }
}

File: src/main/kotlin/com/heybys/oddments/fooddelivery/domain/shop/OptionGroupId.kt
package com.heybys.oddments.fooddelivery.domain.shop

import com.heybys.oddments.base.domain.LongTypeIdentifier
import com.heybys.oddments.base.jpa.hibernate.LongTypeIdentifierJavaType

class OptionGroupId(id: Long) : LongTypeIdentifier(id) {

    class OptionGroupIdJavaType :
        LongTypeIdentifierJavaType<OptionGroupId>(OptionGroupId::class.java)
}
